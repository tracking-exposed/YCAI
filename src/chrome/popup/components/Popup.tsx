import { updateSettings } from '@chrome/dashboard/API/commands';
import { accountSettings } from '@chrome/dashboard/API/queries';
import { ErrorBox } from '@chrome/dashboard/components/common/ErrorBox';
import {
  Button,
  Card,
  CardActions,
  CardContent,
  Divider,
  FormControlLabel,
  Grid,
  makeStyles,
  Switch,
  Typography,
} from '@material-ui/core';
import { Alert, AlertTitle } from '@material-ui/lab';
import * as QR from 'avenger/lib/QueryResult';
import { declareQueries } from 'avenger/lib/react';
import { formatDistance } from 'date-fns';
import parseISO from 'date-fns/parseISO';
import { pipe } from 'fp-ts/lib/function';
import React from 'react';
import { config } from '../../../config';
import Settings from './Settings';

const useStyles = makeStyles((theme) => ({
  container: {
    padding: theme.spacing(2),
    maxWidth: 300,
  },
  content: {
    marginBottom: theme.spacing(2),
  },
  header: {
    marginBottom: theme.spacing(2),
  },
  img: {
    width: '100%',
    maxWidth: 200,
  },
  link: {
    color: 'black',
    textDecoration: 'none',
  },
}));

const PopupLoader: React.FC = () => {
  return (
    <Alert severity="info">
      <AlertTitle>Loading settings...</AlertTitle>
      <strong>
        Access{' '}
        <a href="https://www.youtube.com" target="_blank" rel="noreferrer">
          yutube.com
        </a>
        .
      </strong>
    </Alert>
  );
};

const withQueries = declareQueries({ settings: accountSettings });

export const Popup = withQueries(({ queries }) => {
  const classes = useStyles();

  const version = config.REACT_APP_VERSION;
  const timeago = formatDistance(
    parseISO(config.REACT_APP_BUILD_DATE),
    new Date(),
    {
      addSuffix: true,
    }
  );

  return pipe(
    queries,
    QR.fold(
      () => <PopupLoader />,
      ErrorBox,
      ({ settings }) => {
        return (
          <Card className={classes.container}>
            <CardContent className={classes.content}>
              <Grid className={classes.header} container alignItems="center">
                <Grid item sm={8} xs={8}>
                  <a
                    className={classes.link}
                    href={config.REACT_APP_WEB_URL}
                    target="_blank"
                    rel="noreferrer"
                  >
                    <img className={classes.img} src="/ycai-logo.png" />
                  </a>
                </Grid>
                <Grid item sm={4} xs={4}>
                  <FormControlLabel
                    control={
                      <Switch
                        checked={settings.active}
                        size="small"
                        onChange={(e, c) =>
                          updateSettings({ ...settings, active: c })()
                        }
                      />
                    }
                    label="Enable"
                    labelPlacement="end"
                  />
                </Grid>
                <Grid item xs={12}>
                  <Typography variant="caption">
                    version {version}, released {timeago}
                  </Typography>
                </Grid>
              </Grid>
              <Divider />
              <Settings settings={settings} />
            </CardContent>

            <CardActions>
              <Button
                size="medium"
                color="primary"
                variant="contained"
                href={'/index.html'}
                target="_blank"
                fullWidth
              >
                Dashboard
              </Button>
            </CardActions>
          </Card>
        );
      }
    )
  );
});
