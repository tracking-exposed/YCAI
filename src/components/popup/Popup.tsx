import { updateSettings } from 'API/commands';
import { accountSettings } from 'API/queries';
import { ErrorBox } from '../common/ErrorBox';
import {
  Button,
  Card,
  CardActions,
  CardContent,
  FormControlLabel,
  Grid,
  makeStyles,
  Switch,
  Typography,
} from '@material-ui/core';
import { Alert, AlertTitle } from '@material-ui/lab';
import * as QR from 'avenger/lib/QueryResult';
import { declareQueries } from 'avenger/lib/react';
import { formatDistance } from 'date-fns';
import parseISO from 'date-fns/parseISO';
import { pipe } from 'fp-ts/lib/function';
import React from 'react';
import { config } from '../../config';
import Settings from './Settings';
import { useTranslation } from 'react-i18next';

const useStyles = makeStyles((theme) => ({
  container: {
    padding: theme.spacing(2),
    maxWidth: 300,
  },
  content: {
    marginBottom: theme.spacing(2),
  },
  header: {
    marginBottom: theme.spacing(2),
  },
  img: {
    width: '100%',
    maxWidth: 200,
    display: 'block'
  },
  link: {
    color: 'black',
    textDecoration: 'none',
    display: 'block'
  },
  switchFormControl: {
    margin: 0
  }
}));

const PopupLoader: React.FC = () => {
  return (
    <Alert severity="info">
      <AlertTitle>Loading settings...</AlertTitle>
      <strong>
        Access{' '}
        <a href="https://www.youtube.com" target="_blank" rel="noreferrer">
          yutube.com
        </a>
        .
      </strong>
    </Alert>
  );
};

const withQueries = declareQueries({ settings: accountSettings });

export const Popup = withQueries(({ queries }) => {
  const { t } = useTranslation();
  const classes = useStyles();

  const version = config.REACT_APP_VERSION;
  const timeago = formatDistance(
    parseISO(config.REACT_APP_BUILD_DATE),
    new Date(),
    {
      addSuffix: true,
    }
  );

  return pipe(
    queries,
    QR.fold(
      () => <PopupLoader />,
      ErrorBox,
      ({ settings }) => {
        return (
          <Card className={classes.container}>
            <CardContent className={classes.content}>
              <Grid className={classes.header} container alignItems="center">
                <Grid item xs={7}>
                  <a
                    className={classes.link}
                    href={config.REACT_APP_WEB_URL}
                    target="_blank"
                    rel="noreferrer"
                  >
                    <img className={classes.img} src="/ycai-logo.png" />
                  </a>
                </Grid>

                <Grid item xs={1} />
                <Grid item xs={4} justifyContent="center">
                  <FormControlLabel className={classes.switchFormControl}
                    control={
                      <Switch
                        color="primary"
                        checked={settings.active}
                        size="small"
                        color="primary"
                        onChange={(e, c) =>
                          updateSettings({ ...settings, active: c })()
                        }
                      />
                    }
                    label="Enable" 
                    labelPlacement="end"
                    style={{ background: 'red' }}
                  />
                </Grid>
                <Grid item xs={12}>
                  <Typography variant="caption">
                    {t('popup:version', { version, date: timeago })}
                  </Typography>
                </Grid>
              </Grid>
              <Settings settings={settings} />
            </CardContent>

            <CardActions>
              <Button
                size="medium"
                color="primary"
                variant="contained"
                href={'/index.html'}
                target="_blank"
                fullWidth
              >
                {t('dashboard:title')}
              </Button>
            </CardActions>
          </Card>
        );
      }
    )
  );
});
