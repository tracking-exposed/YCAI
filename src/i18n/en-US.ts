import { CustomTypeOptions } from 'react-i18next';

const resources: CustomTypeOptions['resources'] = {
  title: 'YouChoose AI',
  common: {
    coming_soon: 'Coming soon',
    empty_list: 'No {{resource}} found.',
  },
  actions: {
    popup_bootstrap: 'Bootstrap',
    add: 'Add',
    manage_recommendations: 'Manage recommendations',
    link_channel: 'Link channel',
    unlink_channel: 'Unlink channel',
    delete: 'Delete',
    clear: 'Clear',
    editThisVideo: 'Edit this video',
    importVideos: 'Import videos',
    addToCurrentVideo: 'Add to current video',
    removeFromCurrentVideo: 'Remove from current video',
    compare: 'Compare',
    related: 'Related',
    verify_channel: 'Verify channel',
    generate_keypair: 'Generate keypair',
    refresh_keypair: 'Refresh keypair',
    delete_keypair: 'Delete keypair',
    download_keypair: 'Download keypair',
    pull_creator_videos: 'Import your videos',
    update_creator_videos_list: 'Update your list of videos',
    copy_verification_code: 'Copy code',
    unlink_profile: 'Unlink profile',
    edit_access_token: 'Edit Access Token',
    download_access_token: 'Download Access Token',
    download: 'Download'
  },
  routes: {
    studio: 'Studio',
    statistics: 'Statistics',
    settings: 'Settings',
    link_account: 'Link your account',
  },
  account: {
    channel: 'Your channel',
  },
  creator: {
    title: 'Creator',
  },
  link_account: {
    title: 'Link your account',
    subtitle: 'Our system will verify that you own the channel on this browser',
    label: 'Link your account to start choosing recommendations',
    verification_code_label: 'Verification code',
    verification_code_hint:
      "Copy the code below and add it to your YouTube channel's description",
  },
  youtube: {
    title: 'Youtube',
  },
  recommendations: {
    added_to_video_title: 'Recommendations added',
    by_creator_title: 'Author\'s recommendations',
    total: 'Total recommendations',
    url: 'Recommendation url',
    yours: 'Your recommendations',
    no_items: 'No recommendations found',
    add_to_video: 'Add a recommendation to this video',
    url_placeholder: 'https://youtube.com/watch?v=xxxxx',
    url_helper_text: 'Insert a link',
  },
  dashboard: { title: 'Dashboard' },
  popup: { version: 'version {{version}} build {{data}}' },
  statistics: {
    title: 'Statistics',
    subtitle: 'Statistics computed with resources from other users',
    recommendability_score_title: 'Recommendability Score',
    recommendability_score_subtitle:
      'Where your videos appears as recommended?',
    total_views: 'Total Views',
    total_recommendations: 'Total Recommendations',
    evidences_title: 'Evidences',
    notifications_title: 'Notifications',
    top_n_cc_related_to_your_channel:
      'Top {{count}} CC related to your channel',
    advertising_connected_to_your_videos:
      'Advertising connected to your videos',
  },
  videos: {
    no_results: 'No videos found.',
    no_selected: 'No video selected',
    no_video_id: 'No video id found',
  },
  settings: {
    contentCreatorRecommendationLabel: 'Content Creators',
    contentCreatorRecommendationHint: 'See suggestions by real authors',
    communityRecommendationsLabel: 'Community',
    communityRecommendationsHint: 'Coming soon ðŸŒ»',
    contributeToIndipendentStatsLabel: 'Independent stats',
    contributeToIndipendentStatsHint:
      'Donate anonymously what Youtube recommends and advertises you',
    api_list_title: 'API List',
    encrypted_contributions_private_key: 'You can download your private key in dashboard.',
    keypair_passphrase: 'Key pair passphrase',
    keypair_private_key: 'Key pair secret key',
    keypair_public_key: 'Key pair public key',
    keypair_title: 'Keypair for independent contribution',
    access_token_title: 'You Access Token',
    access_token: 'Access Token'
  },
  ytVideoPage: {
    firstTab: 'Creator Recommendations',
    secondTab: 'Community Recommendations',
    thirdTab: 'Youtube Recommendations',
  },
};

export default resources;
