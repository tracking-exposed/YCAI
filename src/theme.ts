import { createTheme } from '@material-ui/core/styles';

const pink = '#E33180';
const green = '#017374';
const white = "#FFFFFF";

export const YCAITheme = createTheme({
  typography: {
    fontFamily: 'Trex',
    fontWeightRegular: 400,
    fontWeightMedium: 600,
    fontWeightBold: 800,
    // h1: {
    //   fontWeight: 600
    // },
    // h4: {
    //   fontWeight: 600,
    // },
    // h6: {
    //   fontWeight: 600
    // }
  },
  overrides: {
    MuiTabs: {
      root: {
        background: pink,
      },
    },
  },
  palette: {
    background: {
      default: white
    },
    primary: {
      light: '#FF338F',
      main: '#e33180',
      dark: '#C9065E',
      contrastText: '#FFFFFF',
    },
    secondary: {
      light: green,
      main: green,
      dark: green,
      contrastText: white
    }
  },
});

export type YCAITheme = typeof YCAITheme;
