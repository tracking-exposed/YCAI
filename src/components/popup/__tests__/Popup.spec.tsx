import React from 'react';

import '../../../i18n';
import { chrome } from 'jest-chrome';
import renderer from 'react-test-renderer';
import { Popup } from '../Popup';
import { ThemeProvider } from '@mui/material/styles';
import { YCAITheme } from '../../../theme';

const onMessageListener = jest.fn().mockImplementation((r, s, sendRes) => {
  sendRes({ active: true });
  return true;
});

describe('Popup', () => {
  test('Should mount the Popup', () => {
    chrome.runtime.onMessage.addListener(onMessageListener);
    const component = renderer.create(
      <ThemeProvider theme={YCAITheme}>
        <Popup />
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
