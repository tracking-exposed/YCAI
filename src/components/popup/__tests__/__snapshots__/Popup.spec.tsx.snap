// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup Should mount the Popup 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root makeStyles-container-1 css-2f4h8i-MuiPaper-root-MuiCard-root"
>
  <div
    className="makeStyles-header-3 MuiBox-root css-1l4w6pd"
  >
    <a
      rel="noreferrer"
      target="_blank"
    >
      <img
        className="makeStyles-img-5"
        src="/ycai-logo.svg"
      />
    </a>
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard css-1if5c1q-MuiPaper-root-MuiAlert-root"
    role="alert"
  >
    <div
      className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="InfoOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
        />
      </svg>
    </div>
    <div
      className="MuiAlert-message css-acap47-MuiAlert-message"
    >
      <div
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-1ce4ebh-MuiTypography-root-MuiAlertTitle-root"
      >
        Loading settings...
      </div>
      <strong>
        Access
         
        <a
          href="https://www.youtube.com"
          rel="noreferrer"
          target="_blank"
        >
          youtube.com
        </a>
        .
      </strong>
    </div>
  </div>
</div>
`;
